File: 1752-check-if-array-is-sorted-and-rotated/1752-check-if-array-is-sorted-and-rotated.js
/**
 * @param {number[]} nums
 * @return {boolean}
 */
var check = function (nums) {
  nums.push(nums[0])
  for (let i = 1, fold = false; i < nums.length; i++) {
    if (nums[i] < nums[i - 1]) {
      if (fold) return false;
      fold = true;
    }
  }
  return true;
};

File: 1752-check-if-array-is-sorted-and-rotated/1752-check-if-array-is-sorted-and-rotated.go
func check(nums []int) bool {
  nums = append(nums, nums[0])
  for i, fold := 1, false; i < len(nums); i++ {
    if nums[i] < nums[i-1] {
      if fold {
        return false
      }
      fold = true
    }
  }
  return true
}